libs/utils/src/http/error.rs

```diff
@@ -74,6 +77,10 @@ impl ApiError {
                 err.to_string(),
                 StatusCode::REQUEST_TIMEOUT,
             ),
+            ApiError::Cancelled => HttpErrorBody::response_from_msg_and_status(
+                "Request cancelled".to_string(),
```

Comment:
```suggestion
                self.to_string(),
```
That error message is already in the enum variant.

Sorry if I'm meddling in this PR. Just got interested by the project and ended up here